applications/nrf5340_audio/src/modules/display/tab_menu.c

```diff
@@ -0,0 +1,353 @@
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_VOLUME_UP;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void volume_down_button_event_cb(lv_event_t *e)
+
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_VOLUME_DOWN;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void btn4_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_4;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void btn5_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_5;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void vu_bar_update(int vu_val)
+{
+	lv_bar_set_value(vu_bar, vu_val, LV_ANIM_ON);
+}
+
+static int streaming_state_update(lv_obj_t *play_pause_button)
+{
+	int ret;
+	const struct zbus_channel *chan;
+
+	ret = zbus_sub_wait(&le_audio_evt_sub_display, &chan, K_NO_WAIT);
+
+	if (ret == -ENOMSG) {
+		return 0;
+	} else if (ret != 0) {
+		return ret;
+	}
+
+	struct le_audio_msg msg;
+
+	ret = zbus_chan_read(chan, &msg, K_NO_WAIT);
+
+	if (ret) {
+		return ret;
+	}
+	audio_streaming_event = msg.event;
+
+	switch (audio_streaming_event) {
+	case LE_AUDIO_EVT_STREAMING:
+		lv_label_set_text(play_pause_label, LV_SYMBOL_PAUSE);
+		break;
+
+	case LE_AUDIO_EVT_NOT_STREAMING:
+		lv_label_set_text(play_pause_label, LV_SYMBOL_PLAY);
+		break;
+	default:
+		break;
+	}
+	return 0;
+}
+
+static void volume_label_update(lv_obj_t *volume_label)
+{
+	double volume_level_int;
+	char volume_level_str[32];
+
+	volume_level_int = hw_codec_volume_get();
+	sprintf(volume_level_str, "%.0lf", floor((volume_level_int / 128) * 100));
+	strcat(volume_level_str, "%");
+	lv_label_set_text(volume_level_label, volume_level_str);
+}
+
+uint8_t tab_menu_update(uint8_t update_counter)
+{
+	update_counter++;
+
+	if (update_counter >= 5) {
+		if (CONFIG_AUDIO_DEV == HEADSET) {
+			volume_label_update(volume_level_label);
+			if (audio_streaming_event == LE_AUDIO_EVT_STREAMING) {
+				switch (channel) {
+				case AUDIO_CH_L:
+					frame_val = pcsm_avg_frame_val_left_get();
+					break;
+				case AUDIO_CH_R:
+					frame_val = pcsm_avg_frame_val_right_get();
+					break;
+				default:
+					break;
+				}
+				int frame_val_percent =
+					(int)round((double)frame_val / AVG_FRAME_VAL_MAX * 100);
+				vu_bar_update(frame_val_percent);
+			} else {
+				vu_bar_update(0);
+			}
+		}
+		streaming_state_update(play_pause_label);
```

Comment:
```suggestion
		streaming_state_update(play_pause_label, &audio_streaming_event);
```

To reduce the amount of global variables I think it would be better to have the `audio_streaming_event` in this function only. You can then pass it to `streaming_state_update` which will return the current state via the argument. `streaming_state_update` can have a static variable within the function scope to retain the last state across function calls.